### ISSUE TITLE
Network protocol keeps requesting data the peer does not have

### ISSUE THREAD

**Issue by gerardsn** [nuts-foundation/nuts-node#2138]
Network protocol V2's `handleTransactionList` immediately sends a `State` state message to retry set reconciliation if it encounters the `dag.ErrPreviousTransactionMissing` when trying to add a transaction to the DAG.

It's unlikely that this specific peer is going to send a different result than last time (its DB is corrupt? attacker?), so it's going to fail again.

This generates a lot of traffic since there is no delay between the two attempts. A solution is not obvious to me since adding a delay is just reducing the amount of spam a bit, but permanently blocking is not an option either.

**Comment by woutslakhorst** [comment]
It might be best to stop processing when a transaction is missing? Most likely you'll just get it from a different node.

**Comment by gerardsn** [comment]
I agree. This is a very specific example, but requesting the same data again is unlikely to result in a different outcome. The only plausible situation I can think of is when a DID needed to validate a transaction has been added to the DAG, but not yet processed by the VDR.

Stopping the conversation/reconciliation adds a retry interval equal to the length of the gossip interval.

**Comment by woutslakhorst** [comment]
We'll need to change RFC017 first...
nuts-foundation/nuts-specification#262
